plugins/explore/src/plugin.ts

```diff
@@ -16,21 +16,47 @@
 
 import { exploreToolsConfigRef } from '@backstage/plugin-explore-react';
 import { catalogEntityRouteRef, exploreRouteRef } from './routes';
-import { exampleTools } from './util/examples';
-import { createApiFactory, createPlugin } from '@backstage/core-plugin-api';
+import {
+  createApiFactory,
+  createPlugin,
+  discoveryApiRef,
+  fetchApiRef,
+} from '@backstage/core-plugin-api';
+import { ExploreClient, exploreApiRef } from './api';
+// import { exampleTools } from './util/examples';
 
 /** @public */
 export const explorePlugin = createPlugin({
   id: 'explore',
   apis: [
-    // Register a default for exploreToolsConfigRef, you may want to override
-    // the API locally in your app.
+    createApiFactory({
+      api: exploreApiRef,
+      deps: {
+        discoveryApi: discoveryApiRef,
+        fetchApi: fetchApiRef,
+        exploreToolsConfig: exploreToolsConfigRef,
+      },
+      factory: ({ discoveryApi, fetchApi, exploreToolsConfig }) =>
+        // NOTE: The exploreToolsConfig is for backwards compatibility and will be removed in the future
+        new ExploreClient({ discoveryApi, fetchApi, exploreToolsConfig }),
+    }),
+    /**
+     * @deprecated Use ExploreApi from `@backstage/plugin-explore` instead
+     *
+     * Register a default for exploreToolsConfigRef, you may want to override
+     * the API locally in your app.
+     */
     createApiFactory({
       api: exploreToolsConfigRef,
       deps: {},
       factory: () => ({
         async getTools() {
-          return exampleTools;
+          // TODO: Determine if this should return undefined or exampleTools
+          // - `undefined` will enable the explore-backend to be used via the ExploreClient backwards compatibility
+          //    which is likely the desired behavior for anyone that has not yet overriden this API
+          return undefined as any;
```

Comment:
I think undefined makes sense here right? So that `ExploreClient` behaves correctly out of the box. Like you say, anyone using this in production would have been overriding this either way, I think it's fine to not consider the need for a backend in the default case a breaking change.

Reply:
Yeah, that was my thinking. The `undefined` behaviour will enable using the backend by default. If any adopter has defined their own tools it will use that data instead. The only case that will break is if an adopter is using the `explore` plugin and hasn't configured the backend or customized the list of tools. Which is noted in the changeset. If we agree this makes sense, I'll go ahead and remove the todo and finalize the change.